packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2394,20 +2402,19 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
           return;
         }
         assert(_currentBottomSheet!._widget == bottomSheet);
-        if (!isPersistent && !removedEntry) {
-          assert(entry != null);
-          entry!.remove();
-          removedEntry = true;
-        }
       },
       onDismissed: () {
         if (_dismissedBottomSheets.contains(bottomSheet)) {
           setState(() {
             _dismissedBottomSheets.remove(bottomSheet);
           });
         }
+        setState(() {
+          _currentBottomSheet = null;
```

Comment:
We should remove the widget from the tree when dismissing the sheet by gesture, otherwise will be a small leaking.